examples/llm_inference/hugging_face_js/gemma_3n/index.js

```diff
@@ -0,0 +1,387 @@
+);
+const recordButton = document.getElementById(
+  'record-button',
+);
+const sendButton = document.getElementById('send-button');
+const recordButtonIcon = recordButton.querySelector('i');
+const loaderOverlay = document.getElementById('loader-overlay');
+const progressBarFill = document.getElementById('progress-bar-fill');
+const signInMessage = document.getElementById('sign-in-message');
+const loaderMessage = document.getElementById('loader-message');
+
+// --- State Management ---
+let isRecording = false;
+let isLoading = false;
+let mediaRecorder = null;
+let audioChunks = [];
+
+// --- Model-specific constants ---
+// If the user wants to try running on a more limited device, they can switch
+// the demo from default E4B to E2B by appending '?e2b' to the URL.
+const thisUrl = new URL(window.location.href);
+const use_e4b = !thisUrl.searchParams.has('e2b');
+const cacheFileName = use_e4b ? "3n_e4b" : "3n_e2b";
+const remoteFileUrl = use_e4b ? 'https://huggingface.co/google/gemma-3n-E4B-it-litert-lm/resolve/main/gemma-3n-E4B-it-int4-Web.litertlm'
+                              : 'https://huggingface.co/google/gemma-3n-E2B-it-litert-lm/resolve/main/gemma-3n-E2B-it-int4-Web.litertlm';
+// Model size in bytes for reliable progress indication; just hard-coded for now
+const modelSize = use_e4b ? 4274978816 : 3038117888;
+
+// --- Core Functions ---
+
+/**
+ * Updates the progress bar's width.
+ * @param {number} percentage The progress percentage (0-100).
+ */
+function updateProgressBar(percentage) {
+  if (progressBarFill) {
+    progressBarFill.style.width = `${percentage}%`;
+  }
+}
+
+/**
+ * Initializes our local LLM from a StreamReader.
+ */
+let llmInference;
+async function initLlm(modelReader) {
+  console.log('Initializing LLM');
+  loaderMessage.textContent = "Initializing model...";
+
+  // We have no actual progress updates for this last initialization step, but
+  // it's relatively short (<10s on a decent laptop). So we just set to 90%.
+  // TODO: It'd look nicer to have this go from 0 to 100% instead.
+  updateProgressBar(90);
+  const genaiFileset = await FilesetResolver.forGenAiTasks(
+      'https://cdn.jsdelivr.net/npm/@mediapipe/tasks-genai/wasm');
+  LlmInference
+      .createFromOptions(genaiFileset, {
+        baseOptions: {modelAssetBuffer: modelReader},
+        maxTokens: 2048,
+        maxNumImages: 1,
+        supportAudio: true,
+      })
+      .then(llm => {
+        llmInference = llm;
+        // Enable demo now that loading has fully finished.
+        loaderOverlay.style.opacity = '0';
+        setTimeout(() => {
+          loaderOverlay.style.display = 'none';
+          promptInputElement.disabled = false;
+          sendButton.disabled = false;
+          recordButton.disabled = false;
+        }, 300);
+      })
+      .catch((error) => {
+        console.error('Failed to initialize the LLM', error);
+        loaderOverlay.style.display = 'none'; // Hide loader on error
+      });
```

Comment:
Nit: Prefer `await` syntax instead of `.then` since this is an async function.

Reply:
Done.